game/core/game_state.py

```diff
@@ -49,28 +66,61 @@ def get_layer_index(self, name) -> int | None:  # type: ignore[no-untyped-def]
                 return idx
         return None
 
+    @beartype
+    def set_player_data(self) -> None:
+        self.center_x = self.player_data["x"]
+        self.center_y = self.player_data["y"]
+        self.inventory = self.load_inventory(self.player_data["inventory"])
+        self.item = self.load_item(self.player_data["item"])
+        self.vehicle = None
+        if self.player_data["vehicle_type"]:
+            self.vehicle = self.load_vehicle(self.player_data)
+            self.vehicle_x = self.player_data["vehicle_x"]
+            self.vehicle_y = self.player_data["vehicle_y"]
+            self.vehicle_docked = self.player_data["vehicle_docked"]
```

Comment:
We use the term `vehicle`, but this only ever refers to a boat. Maybe we could do a find+replace with `boat`, `watercraft`, or something else to indicate that it is something that can only move on water

Reply:
I was thinking the game itself could potentially have land vehicles in the future, although the implementation would need to be adjusted (eg. `vehicle_docked` would be part of some `vehicle_metadata`